snapcraft/internal/pluginhandler.py

```diff
@@ -656,15 +656,27 @@ def _migrate_files(snap_files, snap_dirs, srcdir, dstdir, missing_ok=False,
     for directory in snap_dirs:
         src = os.path.join(srcdir, directory)
         dst = os.path.join(dstdir, directory)
+
+        dir_stat = os.stat(src, follow_symlinks=follow_symlinks)
+        uid = dir_stat.st_uid
+        gid = dir_stat.st_gid
         os.makedirs(dst, exist_ok=True)
+        os.chown(dst, uid, gid, follow_symlinks=follow_symlinks)
         shutil.copystat(src, dst, follow_symlinks=follow_symlinks)
```

Comment:
Done.

On Thu, Aug 11, 2016 at 3:17 PM, Kyle Fazzari <notifications@github.com>
wrote:

> In snapcraft/internal/pluginhandler.py
> <https://github.com/snapcore/snapcraft/pull/723#discussion_r74485576>:
>
> >          os.makedirs(dst, exist_ok=True)
> > +        os.chown(dst, uid, gid, follow_symlinks=follow_symlinks)
> >          shutil.copystat(src, dst, follow_symlinks=follow_symlinks)
>
> Looking good. Though lines 660-665 look ripe for extraction into a
> function that can also be used on 671-679, no?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/snapcore/snapcraft/pull/723/files/0cb43ec7102415bbf13c10a1cdfee18413711b19#r74485576>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAnbyB6CXcm5MOIsOSOz7zaIQ74N85lyks5qe3VLgaJpZM4JiYV0>
> .
>